File: MVVM/Converters/LinearConverter.vb
Imports System.Globalization
Imports System.Windows.Data

Namespace MVVM.Converters

    Public Class LinearConverter
        Implements IValueConverter

        Public Function Convert(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.Convert
            Return Me.A * CType(Value, Double) + Me.B
        End Function

        Public Function ConvertBack(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Return (CType(Value, Double) - Me.B) / Me.A
        End Function

        Public Property A As Double
        Public Property B As Double

    End Class

End Namespace

File: WPF/MVVM/Converters/LinearConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace Ks
{
    namespace Common.MVVM.Converters
    {
        public class LinearConverter : IValueConverter
        {
            public object Convert(object Value, Type TargetType, object Parameter, CultureInfo Culture)
            {
                return (this.A * System.Convert.ToDouble(Value)) + this.B;
            }

            public object ConvertBack(object Value, Type TargetType, object Parameter, CultureInfo Culture)
            {
                return (System.Convert.ToDouble(Value) - this.B) / this.A;
            }

            public double A { get; set; } = 1;
            public double B { get; set; } = 0;
        }
    }
}
